### PR TITLE
Fixing Tests

### PR DESCRIPTION
**Pull request by vcasadei**

### CODE REVIEW

**Comment by vcasadei** [comment]
@ramonza this is the fix for #5

File: graphite_test.go [hunk 1]
```diff
@@ -75,45 +67,75 @@ func startServer(e *Exporter) {
 			os.Exit(1)
 		}
 		// Handle connections in a new goroutine.
-		go handleRequest(conn)
+		if closeConn.Get() {
+			wg.Done()
+			conn.Close()
+			return
```
Comment (ramonza) [inline graphite_test.go]: Do we not need to close the listener here as well?

File: graphite_test.go [hunk 2]
```diff
@@ -75,45 +67,75 @@ func startServer(e *Exporter) {
...
+		}
+		// Make a buffer to hold incoming data.
+		buf := make([]byte, 1024)
+		r := bufio.NewReader(conn)
+
+		// Read the incoming connection into the buffer.
+		reqLen, err := r.Read(buf)
+		data := string(buf[:reqLen])
+
+		switch err {
```
Comment (ramonza) [inline graphite_test.go]: Shouldn't we check err before reading data?

File: graphite_test.go [hunk 3]
```diff
@@ -75,45 +67,75 @@ func startServer(e *Exporter) {
...
+		case nil:
+			output = output + data
+		default:
```
Comment (ramonza) [inline graphite_test.go]: It's not usual to use a switch statement to check errors.

File: graphite_test.go [hunk 4]
```diff
@@ -284,19 +331,26 @@ func TestMetricsPathOutput(t *testing.T) {
...
+	closeConn.Set(false)
 	exporter, err := NewExporter(Options{Namespace: "opencensus"})
 	if err != nil {
 		t.Fatalf("failed to create graphite exporter: %v", err)
 	}
-
-	closeConn = false
 	output = ""
-	go startServer(exporter)
+	l, err := net.Listen("tcp", exporter.opts.Host+":"+strconv.Itoa(exporter.opts.Port))
```
Comment (ramonza) [inline graphite_test.go]: This code is repeated many times, can we extract a helper function?

File: graphite_test.go [hunk 5]
```diff
@@ -284,19 +331,26 @@ func TestMetricsPathOutput(t *testing.T) {
...
+	if err != nil {
+		fmt.Println("Error listening:", err.Error())
+		return
```
Comment (ramonza) [inline graphite_test.go]: Is return here appropriate? Should we not fail the test?

File: internal/client/client_test.go [hunk 6]
```diff
@@ -33,62 +33,81 @@ const (
...
-var closeConn = false
+var closeConn = &SafeBool{}
 
-func startServer() {
-	l, err := net.Listen("tcp", graphiteHost+":"+strconv.Itoa(graphitePort))
-	if err != nil {
-		fmt.Println("Error listening:", err.Error())
-		return
-	}
+type SafeBool struct {
```
Comment (ramonza) [inline internal/client/client_test.go]: Using package sync/atomic would also work here

File: internal/client/client_test.go [hunk 7]
```diff
@@ -97,13 +116,59 @@ func TestNewGraphite(t *testing.T) {
...
+	var wg sync.WaitGroup
+	output := ""
+
+	l, err := net.Listen("tcp", graphiteHost+":"+strconv.Itoa(graphitePort))
+	if err != nil {
+		fmt.Println("Error listening:", err.Error())
+		return
+	}
+	go func() {
+		wg.Add(1)
```
Comment (ramonza) [inline internal/client/client_test.go]: All this code is repeated several times, is it really necessary to repeat this?

**Comment by ramonza** [comment]
Thanks @vcasadei, I think there are some code quality issue would you mind taking a look?

**Comment by vcasadei** [comment]
@ramonza The PR was not ready for review/merge. Now it is.

File: graphite_test.go (hunk 4 above)
Comment (vcasadei) [reply graphite_test.go]: It's done

Comment (ramonza) [reply graphite_test.go]: Maybe I'm missing something but it still looks like it is repeated. Did you push the changes?

**Pull request merged**